tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_deposit_request.py

```diff
@@ -2,292 +2,139 @@
 
     yield from run_deposit_request_processing(spec, state, deposit_request, validator_index)
 
 
 @with_electra_and_later
 @spec_state_test
-@always_bls
-def test_ineffective_deposit_with_previous_fork_version(spec, state):
-    # Since deposits are valid across forks, the domain is always set with `GENESIS_FORK_VERSION`.
-    # It's an ineffective deposit because it fails at BLS sig verification.
-    # NOTE: it was effective in Altair.
-    assert state.fork.previous_version != state.fork.current_version
-
-    yield from run_deposit_request_processing_with_specific_fork_version(
-        spec,
-        state,
-        fork_version=state.fork.previous_version,
-        effective=False,
-    )
+def test_process_deposit_request_set_start_index(spec, state):
+    assert state.deposit_requests_start_index == spec.UNSET_DEPOSIT_REQUESTS_START_INDEX
 
+    # fresh deposit = next validator index = validator appended to registry
+    validator_index = len(state.validators)
+    # effective balance will be exactly the same as balance.
+    amount = spec.MIN_ACTIVATION_BALANCE
+    deposit_request = prepare_deposit_request(spec, validator_index, amount, signed=True)
 
-@with_electra_and_later
-@spec_state_test
-@always_bls
-def test_effective_deposit_with_genesis_fork_version(spec, state):
-    assert spec.config.GENESIS_FORK_VERSION not in (state.fork.previous_version, state.fork.current_version)
+    yield from run_deposit_request_processing(spec, state, deposit_request, validator_index)
 
-    yield from run_deposit_request_processing_with_specific_fork_version(
-        spec,
-        state,
-        fork_version=spec.config.GENESIS_FORK_VERSION,
-    )
+    assert state.deposit_requests_start_index == deposit_request.index
 
 
 @with_electra_and_later
 @spec_state_test
-def test_success_top_up_to_withdrawn_validator(spec, state):
-    validator_index = 0
+def test_process_deposit_request_set_start_index_only_once(spec, state):
+    initial_start_index = 1
 
-    # Fully withdraw validator
-    set_validator_fully_withdrawable(spec, state, validator_index)
-    assert state.balances[validator_index] > 0
-    next_epoch_via_block(spec, state)
-    assert state.balances[validator_index] == 0
-    assert state.validators[validator_index].effective_balance > 0
-    next_epoch_via_block(spec, state)
-    assert state.validators[validator_index].effective_balance == 0
+    # fresh deposit = next validator index = validator appended to registry
+    validator_index = len(state.validators)
+    # effective balance will be exactly the same as balance.
+    amount = spec.MIN_ACTIVATION_BALANCE
+    deposit_request = prepare_deposit_request(spec, validator_index, amount, signed=True)
 
-    # Make a top-up balance to validator
-    amount = spec.MAX_EFFECTIVE_BALANCE // 4
-    deposit_request = prepare_deposit_request(spec, validator_index, amount, len(state.validators), signed=True)
+    state.deposit_requests_start_index = initial_start_index
```

Comment:
We should add an assert that `deposit_request.index` is different than `initial_start_index`:
```suggestion
    assert initial_start_index != deposit_request.index
    state.deposit_requests_start_index = initial_start_index
```

If they're the same, the test doesn't tell us anything.